api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -152,4 +153,11 @@ static StructType getType(StructType partitionType) {
    * are determined by these offsets. The returned list must be sorted in ascending order.
    */
   List<Long> splitOffsets();
+
+  /**
+   * @return The sequence number to identify the order in which data files and deletion files are to be processed.
+   * If the sequence number is not specified it is inherited from the manifest file struct in the manifest list file.
+   */
+  Long sequenceNumber();
```

Comment:
Should this ever return null? I would rather use a valid number and a primitive type to ensure that we _always_ have a sequence number that is valid. For older data, the sequence number should be 0, indicating that it is older than when we started writing sequence numbers.

Reply:
I use boxing here because I don't want to rewrite the manifest file. The sequence number is not determined when committing the update, leave it to null so that we can inherit the sequence number from the value in the manifest list file.

If we use the primitive type, we need to check where the sequence number of the manifest is null or not.  It should be doable, let me update this.